File: LeetCode/Medium/0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public static int minSubArrayLen(int target, int[] nums) {
        Deque<Integer> q = new ArrayDeque<>();
        int n = nums.length;
        int len = n + 1;
        int sum = nums[0];

        int cur = 0;
        q.add(nums[0]);

        while (cur < n) {
            if (sum == target) {
                len = Math.min(len, q.size());
                sum -= q.getFirst();
                q.removeFirst();
                cur += 1;
                if (cur < n) {
                    q.addLast(nums[cur]);
                    sum += nums[cur];
                }
            } else if (sum > target) {
                len = Math.min(len, q.size());
                sum -= q.getFirst();
                q.removeFirst();
            } else {
                cur += 1;
                if (cur < n) {
                    q.addLast(nums[cur]);
                    sum += nums[cur];
                }
            }

        }
        if (len == n + 1) {
            return 0;
        }
        return len;
    }
}

File: LeetCode/Medium/0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
    // sort.Ints(nums)

    l := 0
    r := 0
    s := nums[l]

    n := len(nums)

    ans := n + 1

    // 1, 2, 2, 3, 3, 4
    fmt.Println(nums)
    for (r != n) {
        if (s == target) {
            fmt.Println(l, r, s)
            ans = min(ans, r - l + 1)
            r += 1
            if (r < n) {
                s += nums[r]
            }
        } else if (s < target) {
            r += 1
            if (r < n) {
                s += nums[r]
            }
        } else {
            fmt.Println(l, r, s)

            ans = min(ans, r - l + 1)
            s -= nums[l]
            l += 1
        }
    }


    if (ans == n + 1) {
        return 0
    }

    return ans
}